src/main.rs

```diff
@@ -488,7 +527,6 @@ impl FileReceiver {
             self.offset += chunk.data.len() as u64;
         }
 
-        self.out_file = Some(out_file);
```

Comment:
We should preserve this one! Without it, we’re still re-opening the file for every chunk received!